### PR TITLE
Adding WandB Artifacts

### PR DESCRIPTION
**Pull request by digantamisra98**
Adding W&B Artifacts functionality along with external storage bucket reference. This is WIP and might not work atm. Next plan is to add a metric for capturing per_exp weights of the model.

### CODE REVIEW

**Comment by AndreaCossu** [comment]
Thanks @digantamisra98 ! Let me know when this is ready to be reviewed :smile:

**Comment by digantamisra98** [comment]
@AndreaCossu sure yes.

**Comment by digantamisra98** [comment]
@ggraffieti @AndreaCossu What should be the tests that I should define for the new metric I have introduced?

File: avalanche/logging/wandb_logger.py [hunk 1]
```diff
@@ -4,48 +4,80 @@
...
         """
         Creates an instance of the `WandBLogger`.
-
-        :param project_name: Name of the W&B project.:
-        :param run_name: Name of the W&B run.:
+        :param project_name: Name of the W&B project.
+        :param run_name: Name of the W&B run.
+        :param log_artifacts: Option to log model weights as W&B Artifacts.
+        :param path: Path to locally save the model checkpoints.
+        :param uri: Reference to external URI.
```
Comment (AndreaCossu) [inline avalanche/logging/wandb_logger.py:60]: could you expand a bit on this parameter?

Comment (digantamisra98) [reply avalanche/logging/wandb_logger.py:60]: The `uri` parameter?

Comment (AndreaCossu) [reply avalanche/logging/wandb_logger.py:60]: exactly

Comment (digantamisra98) [reply avalanche/logging/wandb_logger.py:60]: Sure, I'll add a more detailed definition of that parameter.

**Comment by digantamisra98** [comment]
@AndreaCossu is this okay to merge? I'll do further testing once merged and add docs.

**Comment by AndreaCossu** [comment]
@digantamisra98 most tests are still failing. I think the PR is ready, we can merge it as soon as tests complete successfully.

**Comment by digantamisra98** [comment]
@AndreaCossu the tests are failing because I haven't added any new unit tests for the new metric.

**Comment by AndreaCossu** [comment]
I don't think this is the reason. Since you have not defined tests for new metric, there is nothing that should fail. If you look at the errors inside unit tests you can better see what is going on. For example:
```
  File "/__w/avalanche/avalanche/avalanche/evaluation/metrics/__init__.py", line 95, in <module>
    from checkpoint import *
ModuleNotFoundError: No module named 'checkpoint'
```
This is an import error that is not related to metric tests.

**Comment by digantamisra98** [comment]
Missed that error completely, my bad. Fixing them now.

**Comment by digantamisra98** [comment]
@AndreaCossu all tests now pass, this can be merged now.

**Pull request merged**